### DISCUSSION TITLE
Comparing images by pixel colors

### DISCUSSION THREAD

**Discussion by b1zar** [ImageMagick/ImageMagick#2358] [Help]
Hey. I just started to deal with ImageMagic, because of my knowledge I don't know how to make a percentage comparison of pixel colors. For example, I have two images:
ImageA: https://ibb.co/KGGyyjd
ImageB: https://ibb.co/1KX3ZX6

For my specific tasks, these are two identical images, even those that are located differently. Using the command "magick convert ImageA.png -format %%c histogram:info:> 1.txt" for two images shows that their colors are identical. How can you implement comparisons of pictures by pixel colors, not in the output of a histogram of 100k + lines, but as a percentage?

Let's say there are 2 images of 100 pixels consisting of 3 colors. Histogram output for images:
ImageA:
    30: (0,18,14,255) #00120EFF srgba(0,18,14,1)
    47: (0,19,15,255) #00130FFF srgba(0,19,15,1)
    23: (0,26,17,255) #001A11FF srgba(0,26,17,1)
ImageB:
    31: (0,18,14,255) #00120EFF srgba(0,18,14,1)
    46: (0,19,15,255) #00130FFF srgba(0,19,15,1)
    23: (0,26,17,255) #001A11FF srgba(0,26,17,1)

From the example you can see that the image differs by 2 pixels, respectively 98 pixels are similar, which equals 98%.
How can this be implemented in ImageMagic?

**Comment by snibgo** [comment]
What version of IM, on what platform?

I think the best way is to read the text histograms in a script, and do the calculation.

**Reply by b1zar** [reply]
Sorry, I forgot to indicate. Windows 10 platform, ImageMagick-7.0.10-24-Q16-HDRI-x64-dll

**Reply by b1zar** [reply]
How can I shorten the output "-format %% c histogram: info:" from this format "29: (255,254,253,255) #FFFEFDFF srgba (255,254,253,1)" to "29: #FFFEFDFF". To reduce the processing of information by the script

**Reply by fmw42** [reply]
You cannot change the %c format. You can parse the output to remove information by piping to the unix sed or awk commands. But that is not going to help reduce processing time.

**Reply by b1zar** [reply]
It is not a problem to take the necessary values ​​from the file, just the processing of a 10MB and 2.5 .txt file will differ in speed. I also do not use the Unix environment, but I can imagine how to make a handler

**Comment by fmw42** [comment]
If on a Unix-like OS (Linux, Mac OSX, Windows 10 Unix) see my script histcompare at my web site http://www.fmwconcepts.com/imagemagick/index.php
